ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Protocol/Praos.hs

```diff
@@ -78,31 +78,97 @@ import           Ouroboros.Consensus.Protocol.Abstract
 import           Ouroboros.Consensus.Protocol.Signed
 import           Ouroboros.Consensus.Util.Condense
 
+-- The Praos paper can be located at https://ia.cr/2017/573
+--
+-- A block as defined in Praos (section 3, definition 2, then extended in Fig 9)
+-- consist of a tuple:
+--
+-- > B = (slⱼ, st, d, B_πj, ρ, σⱼ)
+--
+-- where
+-- - slⱼ: the slot at which the block was generated.
+--
+-- - st: the current state.
```

Comment:
Defn 2 from the paper also says `A state is a string st ∈ {0,1}^λ`. ... ah, reading farther along in that paragraph, I see that `st` is the hash of the previous block. That's probably a clarification to make here that'll avoid confusing future reader's confusion.

Reply:
Also, let's call out the connections between these components and fields of `SimpleStdHeader` and/or `SimpleBody` and/or `SimplePraosExt`/`PraosFields` etc where we can. Ideally, these components and those fields will be one-to-one!

Reply:
I added lines specifying how they are named in the code, and clarified the `st` part. Good point! I actually had not map in my mind the state to the hash of the previous block.